src/ui/code/settingsPage.ts

```diff
@@ -3672,6 +3723,46 @@ export class settingsPage {
 				alert("Diagnostics are enabled for this session only. Check this page for more info.");
 				window.open("https://github.com/salarcode/SmartProxy/wiki/Enable-Diagnostics")
 			}
+		},
+		onWindowUnload(event) {
+			// Check GeneralOptions via comparison
+			if (!settingsPage.readGeneralOptions().Equals(settingsPage.currentSettings.options)) {
+				settingsPage.changeTracking.options = true;
+			}
+			if (!settingsPage.changeTracking.isDirty())
+				return;
+
+			// If user choose to stay, show a messagebox asking if saving all unsaved changes.
+			jq(window).one("focus", () => {
+				// setTimeout to avoid when user choose to leave, this messagebox flash before the window close itself
+				setTimeout(() => {
+					messageBox.confirm(api.i18n.getMessage("settingsConfirmSaveAllChanged"),
+						() => {
+							if (settingsPage.changeTracking.options) {
+								settingsPage.uiEvents.onClickSaveGeneralOptions();
+							}
+							if (settingsPage.changeTracking.smartProfiles || settingsPage.changeTracking.rulesSubscriptions) {
+								for (let pageProfile of settingsPage.pageSmartProfiles) {
+									settingsPage.uiEvents.onClickSaveSmartProfile(pageProfile);
+								}
+								if (settingsPage.unsavedProfile) {
+									// if profile name not set, error message box will show.
+									settingsPage.uiEvents.onClickSaveSmartProfile(settingsPage.unsavedProfile);
+								}
+							}
+							if (settingsPage.changeTracking.servers || settingsPage.changeTracking.activeProxy) {
+								settingsPage.uiEvents.onClickSaveProxyServers();
+							}
+							if (settingsPage.changeTracking.serverSubscriptions) {
+								settingsPage.uiEvents.onClickSaveServerSubscriptionsChanges();
+							}
+						});
+				}, 200);
+			});
+
+			// Browser will show a dialog asking user to leave or stay.
+			event.preventDefault();
```

Comment:
Is there a way to customize the confirm message that browsers are showing? I did not see any!